### PR TITLE
Add DeleteSuggestionCommandParser and DeleteSuggestionCommandParserTest

### LINKED ISSUE #260: Create DeleteCommandParser and DeleteCommandParserTest

### PR DESCRIPTION
**Pull request by johannagwan**
Closes #260 

These have similar implementation as `OpenSuggestionCommandParser` and `OpenSuggestionCommandParserTest`

- [x]  `DeleteSuggestionCommandParser` (update based on `OpenSuggestionCommandParser`
- [x] `DeleteSuggestionCommandParserTest` (after the `OpenSuggestionCommandParserTest` is approved)

NOTE: 
- will add more test cases for `OpenSuggestionCommand` in the `SuggestionEngineImplTest` to match `DeleteSuggestionCommand` test cases.
- will abstract out repetitive codes into a test util class

### CODE REVIEW

File: src/main/java/com/notably/logic/parser/suggestion/DeleteSuggestionCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+            throw new ParseException(String.format("Invalid input"));
+        }
+
+        String title = argMultimap.getValue(PREFIX_TITLE).get();
+        AbsolutePath uncorrectedPath = ParserUtil.createAbsolutePath(title, model.getCurrentlyOpenPath());
+        Optional<AbsolutePath> correctedPath = correctionEngine.correct(uncorrectedPath).getCorrectedItem();
+
+        return correctedPath
+                .map(DeleteSuggestionCommand::new)
+                .orElseThrow(() -> new ParseException("Invalid path"));
```
Comment (kevinputera) [inline src/main/java/com/notably/logic/parser/suggestion/DeleteSuggestionCommandParser.java]: Love this usage of functional-style java!

**Review by kevinputera — APPROVED** [review]
LGTM!

File: src/main/java/com/notably/logic/parser/suggestion/DeleteSuggestionCommandParser.java (hunk 1 above)
Comment (johannagwan) [reply src/main/java/com/notably/logic/parser/suggestion/DeleteSuggestionCommandParser.java]: Lovely!

**Review by kevinputera — APPROVED** [review]
LGTM

**Review by kevinputera — APPROVED** [review]
LGTM

**Pull request merged**